src/query_ctx.c

```diff
@@ -62,32 +69,33 @@ void QueryCtx_SetGlobalExecutionCtx(CommandCtx *cmd_ctx) {
 }
 
 void QueryCtx_SetAST(AST *ast) {
-	QueryCtx *ctx = _QueryCtx_GetCtx();
+	QueryCtx *ctx = _QueryCtx_GetCreateCtx();
 	ctx->query_data.ast = ast;
 }
 
 void QueryCtx_SetGraphCtx(GraphContext *gc) {
-	QueryCtx *ctx = _QueryCtx_GetCtx();
+	QueryCtx *ctx = _QueryCtx_GetCreateCtx();
 	ctx->gc = gc;
 }
 
 void QueryCtx_SetResultSet(ResultSet *result_set) {
-	QueryCtx *ctx = _QueryCtx_GetCtx();
+	QueryCtx *ctx = _QueryCtx_GetCreateCtx();
 	ctx->internal_exec_ctx.result_set = result_set;
 }
 
 void QueryCtx_SetLastWriter(OpBase *last_writer) {
-	QueryCtx *ctx = _QueryCtx_GetCtx();
+	QueryCtx *ctx = _QueryCtx_GetCreateCtx();
 	ctx->internal_exec_ctx.last_writer = last_writer;
 }
 
 AST *QueryCtx_GetAST(void) {
 	QueryCtx *ctx = _QueryCtx_GetCtx();
-	return ctx->query_data.ast;
+	if(ctx) return ctx->query_data.ast;
+	return NULL;
 }
 
 rax *QueryCtx_GetParams(void) {
-	QueryCtx *ctx = _QueryCtx_GetCtx();
+	QueryCtx *ctx = _QueryCtx_GetCreateCtx();
```

Comment:
Shouldn't we call `_QueryCtx_GetCtx` here? expecting params to be set?
similar to `QueryCtx_GetAST`